agent/app/agent_capability_test.go

```diff
@@ -394,116 +379,6 @@ func TestCapabilitiesTaskIAMRoleForSupportedDockerVersion(t *testing.T) {
 	assert.True(t, ok, "Could not find iam capability when expected; got capabilities %v", capabilities)
 }
 
-func TestCapabilitiesTaskIAMRoleForUnSupportedDockerVersion(t *testing.T) {
```

Comment:
The unit tests below are deleted because we are no longer querying the docker engine for the list of supported docker api versions, we are advertising our list of "known" api versions that we support. The side-effect of this is we can no longer mock this docker engine API call in our unit tests.

This should be safe as this particular feature began in docker api version 1.22, which is now deprecated and we will no longer support anyways, so the agent would fail long before this capability logic anyways.